examples/vector_tile_3d_mesh_Mapbox.html

```diff
@@ -0,0 +1,131 @@
+            view.tileLayer.noTextureColor = new itowns.THREE.Color(0x95c1e1);
+
+            // Disable atmosphere lighting
+            view.getLayerById('atmosphere').visible = false;
+            view.getLayerById('atmosphere').fog.enable = false;
+
+            // Setup loading screen and debug menu
+            setupLoadingScreen(viewerDiv, view);
+            const debugMenu = new GuiTools('menuDiv', view, 300);
+
+
+
+            // ---------- DISPLAY CONTEXTUAL DATA : ----------
+
+            // Add two elevation layers, each with a different level of detail. Each layer's properties are defined in
+            // a json file.
+            function addElevationLayerFromConfig(config) {
+                config.source = new itowns.WMTSSource(config.source);
+                view.addLayer(new itowns.ElevationLayer(config.id, config))
+                    .then(debugMenu.addLayerGUI.bind(debugMenu));
+            }
+            // itowns.Fetcher.json('./layers/JSONLayers/IGN_MNT_HIGHRES.json').then(addElevationLayerFromConfig);
+            itowns.Fetcher.json('./layers/JSONLayers/WORLD_DTM.json').then(addElevationLayerFromConfig);
+
+            
+            // ---------- DISPLAY ORTHO-IMAGES : ----------
+
+            const ortho = itowns.Fetcher.json('./layers/JSONLayers/Ortho.json')
+                .then(function _(config) {
+                    config.source = new itowns.WMTSSource(config.source);
+                    var layer = new itowns.ColorLayer('Ortho', config);
+                    return view.addLayer(layer);
+                });
+
+            // ---------- DISPLAY VECTOR TILED BUILDING DATA AS 3D MESHES : ----------
+
+            // // Define the source of the building data : those are vector tiled data from the geoportail.
+            const buildingsSource = new itowns.VectorTilesSource({
+                style: 'http://localhost:8081/testMvt.json',
+                accessToken: 'pk.eyJ1IjoidGFtaWQyMTA2OCIsImEiOiJjbGlicGN2bWEwOWNvM2huejBwa2V1cGV0In0._dykOL_KbMkzWw8EXdclFw',
+                filter: (layer) => {
+                    return layer.id === 'building';
+                },
+            });
+
+            // Get buildings ground elevation
+            function altitudeBuildings(properties, coord) {
+                if (coord) {
+                    return itowns.DEMUtils.getElevationValueAt(view.tileLayer, coord);
+                }
+            }
+
+            // Create a FeatureGeometryLayer to support building data.
+            var buildingsLayer = new itowns.FeatureGeometryLayer('VTBuilding',{
+                batchId: function (property, featureId) { return featureId; },
+                source: buildingsSource,
+                zoom: {
+                    min: 15,
+                    // max: 13,
+                },
+                style: {
+                    fill: {
+                        base_altitude: (p, c) => {
+                            return (altitudeBuildings(p, c) || 0) + p.min_height;
```

Comment:
this computation can be done in `altitudeBuildings` directly

Reply:
I separate both to show more easily the part where we get the ground elevation data for a given building. The p.min_height is a specificité of the mapbow streets V8 building tile set (for building parts that are not directly on the ground).

Reply:
ok, I found it clearer to put everything in one function but it's not critical so I'll let you decide